src/main/kotlin/org/rust/lang/core/parser/RustParserUtil.kt

```diff
@@ -226,6 +226,32 @@ object RustParserUtil : GeneratedParserUtilBase() {
     fun isPathMode(b: PsiBuilder, level: Int, mode: PathParsingMode): Boolean =
         mode == getPathMod(b.flags)
 
+    @JvmStatic
+    fun parseFloatLiteral(b: PsiBuilder, level: Int): Boolean {
```

Comment:
Since we collapse tokens by the parser, it breaks proper highlighting of float literals by `RsHighlighter`. More precisely, the color of dot in float literals was blue before your change, now it's black

Reply:
Oh